### PR TITLE
[MM-51844 & MM-51843] Work template permissions check

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
Add more checks around work template permissions in the frontend.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-51844
https://mattermost.atlassian.net/browse/MM-51843

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/selectors/work_template.ts [hunk 1]
```diff
@@ -3,14 +3,23 @@
...
 
 export const areWorkTemplatesEnabled = createSelector(
     'areWorktemplatesEnabled',
     (state: GlobalState) => getFeatureFlagValue(state, 'WorkTemplate') === 'true',
-    (workTemplateFF) => {
-        return workTemplateFF;
+    (state: GlobalState) => getLicense(state),
+    (state: GlobalState) => haveICurrentTeamPermission(state, Permissions.CREATE_PUBLIC_CHANNEL) || haveICurrentTeamPermission(state, Permissions.CREATE_PRIVATE_CHANNEL),
+    (state: GlobalState) => haveICurrentTeamPermission(state, 'playbook_public_create'),
+    (state: GlobalState) => haveICurrentTeamPermission(state, 'playbook_private_create'),
```
Comment (julmondragon) [inline webapp/channels/src/selectors/work_template.ts:18]: These keys are available from Permissions: `PLAYBOOK_PUBLIC_CREATE`, `PLAYBOOK_PRIVATE_CREATE`

File: webapp/channels/src/components/work_templates/components/customize.tsx [hunk 2]
```diff
@@ -38,6 +40,10 @@ const Customize = ({
     const templateHasChannels = template.content.findIndex((item) => item.channel) !== -1;
     const templateHasBoards = template.content.findIndex((item) => item.board) !== -1;
     const templateHasPlaybooks = template.content.findIndex((item) => item.playbook) !== -1;
+    const canCreatePublicChannel = useSelector((state: GlobalState) => haveICurrentTeamPermission(state, Permissions.CREATE_PUBLIC_CHANNEL));
+    const canCreatePrivateChannel = useSelector((state: GlobalState) => haveICurrentTeamPermission(state, Permissions.CREATE_PRIVATE_CHANNEL));
+    const canCreatePublicPlaybook = useSelector((state: GlobalState) => haveICurrentTeamPermission(state, 'playbook_public_create'));
+    const canCreatePrivatePlaybook = useSelector((state: GlobalState) => haveICurrentTeamPermission(state, 'playbook_private_create'));
```
Comment (julmondragon) [inline webapp/channels/src/components/work_templates/components/customize.tsx:46]: Same about the keys.

File: webapp/channels/src/components/work_templates/components/customize.tsx [hunk 3]
```diff
@@ -5,15 +5,17 @@ import React, {useEffect} from 'react';
 import styled from 'styled-components';
 import {useIntl} from 'react-intl';
 import {useSelector} from 'react-redux';
-
```
Comment (julmondragon) [inline webapp/channels/src/components/work_templates/components/customize.tsx:8]: nit: I think this is wanted.

**Review by julmondragon — APPROVED** [review]
Thanks @JulienTant !

File: webapp/channels/src/selectors/work_template.ts (hunk 1 above)
Comment (JulienTant) [reply webapp/channels/src/selectors/work_template.ts:18]: Ohhh I was not available when I made my branch, this probably appeared after my rebase!

File: server/channels/app/worktemplates/templates.yaml [hunk 4]
```diff
@@ -42,7 +42,7 @@ content:
       template: "Product Release"
       name: "Feature release"
       id: product-release-playbook
-      illustration: "/static/worktemplates/product_teams/feature_release/playbook.png"
```
Comment (marianunez) [inline server/channels/app/worktemplates/templates.yaml:45]: Oh sorry, the typo was that it's missing an `s` at the end. The new file I had added is: https://github.com/mattermost/mattermost-server/blob/master/webapp/channels/src/images/worktemplates/product_teams/feature_release/playbooks.png

Comment (JulienTant) [reply server/channels/app/worktemplates/templates.yaml:45]: There was an existing one (same picture, same size) in the `playbooks/` folder. That's a pattern all other templates respect so I made sure to stick to it

Comment (JulienTant) [reply server/channels/app/worktemplates/templates.yaml:45]: Really? I can't see the diff lol

https://github.com/mattermost/mattermost-server/blob/master/webapp/channels/src/images/worktemplates/playbooks/feature_lifecycle.png

https://github.com/mattermost/mattermost-server/blob/master/webapp/channels/src/images/worktemplates/product_teams/feature_release/playbooks.png

File: webapp/channels/src/images/worktemplates/playbooks/feature_lifecycle.png
Comment (marianunez) [inline webapp/channels/src/images/worktemplates/playbooks/feature_lifecycle.png:1]: Now I'm confused, they do look the same hahaa

Comment (JulienTant) [reply webapp/channels/src/images/worktemplates/playbooks/feature_lifecycle.png:1]: I swapped the new picture you added to the `playbooks/` folder

**Review by marianunez — APPROVED** [review]
LGTM, Thanks @JulienTant!

**Review by pvev — APPROVED** [review]
LGTM! thanks @JulienTant :shipit:

**Review by stevemudie — APPROVED** [review]
Looks good @JulienTant!

**Pull request merged**